src/resource/jsonschema/resource.js

```diff
@@ -101,14 +98,15 @@ export default class JSONSchemaResource extends Resource {
         return new JSONSchemaResourceList(this);
     }
 
-    toRef() {
-        return new Reference(this.id, this.registry);
-    }
-
     /* Validate data against this JSONSchema schema.
      */
     validate(data) {
-        const result = this.validator.validate(data, this.schema);
+        if (!this.validator) {
```

Comment:
The validator is now lazy-created. This is probably not necessary, but felt like a good way to support non-DAG references in the future in which referenced types are added *after* resource construction.